src/main/java/core/mvc/DispatcherServlet.java

```diff
@@ -20,52 +27,61 @@
     private static final String BASE_PACKAGE = "next.controller";
 
     private List<HandlerMapping> handlerMappings;
+    private HandlerMethodArgumentResolverComposite handlerMethodArgumentResolverComposite;
 
     @Override
     public void init() {
         final AnnotationHandlerMapping ahm = new AnnotationHandlerMapping(BASE_PACKAGE);
         ahm.initialize();
 
         handlerMappings = Arrays.asList(ahm);
+
+        handlerMethodArgumentResolverComposite = new HandlerMethodArgumentResolverComposite();
+        handlerMethodArgumentResolverComposite.addResolver(new ServletRequestResolver());
+        handlerMethodArgumentResolverComposite.addResolver(new ServletResponseResolver());
+        handlerMethodArgumentResolverComposite.addResolver(new RequestParamResolver());
+        handlerMethodArgumentResolverComposite.addResolver(new ModelAttributeResolver());
+        handlerMethodArgumentResolverComposite.addResolver(new PathVariableResolver());
     }
 
     @Override
     protected void service(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
         logger.debug("Method : {}, Request URI : {}", req.getMethod(), req.getRequestURI());
-
-        final Object handler = getHandler(req);
-        final ModelAndView mav = handle(handler, req, resp);
-        render(mav, req, resp);
-    }
-
-    private ModelAndView handle(Object handler, HttpServletRequest req, HttpServletResponse resp) throws ServletException {
         try {
-            if (handler instanceof HandlerExecution) {
-                return ((HandlerExecution) handler).handle(req, resp);
+            ModelAndView mav = null;
+            try {
```

Comment:
예외처리가 2번 들어가며 조금 복잡해졌네요 
이 부분을 개선해 볼 수 없을까요?

Reply:
총 2개의 `try 문` 중에서 하나를 메서드로 빼서 개선해 보았습니다 !

이것이 기대하신 개선사항이 아닐 것 같은 느낌이 드는데요.. 

혹시 더 깔끔해보이는 방법이 있다면 힌트 좀 주실 수 있나요~? 🙇‍♂

Reply:
음 .. 예를 든다면.. multi catch 를 이용해볼 수 있지 않을까요

```suggestion
ModelAndView mav = null;

try {
    final Object handler = getHandler(req);
    mav = handle(handler, req, resp);
} catch (HandlerNotFoundException handlerNotFoundException) {
    mav = new ModelAndView(new JspView("/error/page404.jsp"));
} catch (Exception e) {
    logger.error("Exception is: ", e);
    throw new ServletException(e.getMessage());
}
    
render(mav, req, resp);
```